finat/hct.py

```diff
@@ -0,0 +1,68 @@
+import numpy
+
+import FIAT
+
+from gem import Literal, ListTensor
+
+from finat.fiat_elements import ScalarFiatElement
+from finat.physically_mapped import PhysicallyMappedElement, Citations
+
+
+class HsiehCloughTocher(PhysicallyMappedElement, ScalarFiatElement):
+    def __init__(self, cell, degree):
+        if degree != 3:
+            raise ValueError("Degree must be 3 for HCT element")
+        if Citations is not None:
+            Citations().register("Clough1965")
+        super().__init__(FIAT.HsiehCloughTocher(cell))
+
+    def basis_transformation(self, coordinate_mapping):
+        # Jacobians at edge midpoints
+        J = coordinate_mapping.jacobian_at([1/3, 1/3])
```

Comment:
That's definitely the jacobian at the cell centre.

Reply:
Also the Jacobian everywhere on the cell if it is a constant. What is the right thng here, Hermite does not assume constant Jacobian, how well does the transformation theory extend to non-constant Jacobians?

Reply:
I recall having some discussions with @rckirby but we never came to a conclusion (nor did we write any maths down). Maybe it's "trivial" and as long as the scaling objects are picked at the appropriate place and the refinement of the mesh converges to affine mapped then everything falls out in the wash.

Reply:
Section 3.4.1 of the original zany element paper says it works, at least for Hermite.  I haven't ever tried this at all, not sure what the delicacies are.